calcite/src/main/java/husky/sql/calcite/planner/HuskyPartsQueryPlanner.java

```diff
@@ -0,0 +1,335 @@
+package husky.sql.calcite.planner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+
+
+import husky.sql.calcite.table.SimplePartsTable;
+import husky.sql.calcite.table.SimplePartsTableFactory;
+import husky.sql.calcite.Exceptions.TableException;
+import husky.sql.calcite.plan.nodes.HuskyConventions;
+import husky.sql.calcite.plan.rules.HuskyRuleSets;
+import org.apache.calcite.config.Lex;
+import org.apache.calcite.jdbc.CalciteConnection;
+import org.apache.calcite.model.ModelHandler;
+import org.apache.calcite.plan.*;
+import org.apache.calcite.plan.hep.HepMatchOrder;
+import org.apache.calcite.plan.hep.HepPlanner;
+import org.apache.calcite.plan.hep.HepProgramBuilder;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.RelRoot;
+import org.apache.calcite.rel.RelVisitor;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql2rel.RelDecorrelator;
+import org.apache.calcite.tools.*;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+import java.io.FileWriter;
+
+import static org.apache.calcite.plan.Contexts.EMPTY_CONTEXT;
+
+public class HuskyPartsQueryPlanner {
```

Comment:
`extends HuskyQueryPlanner`? Do not copy the exact functions.

Reply:
Some methods in HuskyQueryPlanner are declaimed private. The child class cannot access.